# Локальный проект “ТОС”

## Описание проекта

Проект направлен на обработку и визуализацию данных о территориально-обособленных сообществах (ТОС) Белгородского района с использованием отечественных инструментов, таких как DataLens. Целью является создание интерактивного датастори, который позволит анализировать характеристики ТОС, выявлять тренды и предоставлять информацию для заинтересованных сторон.

## Цель проекта

Основная цель проекта — разработка аналитического инструмента для визуализации и анализа данных о ТОС Белгородского района, что позволит повысить эффективность работы общественных активистов и местных сообществ.

## План работы

1. Сбор данных: Подготовка и сбор необходимых данных из файла Excel и других источников.

2. Обработка данных: Очистка и структурирование данных для дальнейшего анализа.

3. Анализ данных: Проведение анализа характеристик ТОС с использованием статистических методов.

4. Визуализация данных: Создание визуальных отчетов с использованием DataLens для представления результатов анализа.

5. Публикация результатов: Публикация интерактивного датастори для доступа к информации о состоянии ТОСов.

6. Аналитический отчет: Подготовка аналитического отчета на основе полученных данных и выводов.



In [1]:
# импортируем необходимые библиотеки
import pandas as pd

In [2]:
# отображаем все столбцы
pd.set_option('display.max_columns', None)  
# устанавливаю формат для цифр
pd.options.display.float_format = '{:.2f}'.format

## Открыть файлы с данными

In [3]:
try:
    tos = pd.read_excel('/Users/sergey/Desktop/Мастерская/tos.xlsx', sheet_name = 'ТОС апр 2023')
    tos_district = pd.read_excel('/Users/sergey/Desktop/Мастерская/tos.xlsx', sheet_name = 'район')
    tos_area = pd.read_excel('/Users/sergey/Desktop/Мастерская/tos.xlsx', sheet_name = 'в области')
except:
    data = None  
    
if tos is None or tos_district is None or tos_area is None:
    print('Что-то не так')
else:
    print('Данные успешно загружены')

Данные успешно загружены


## Предобработка данных

**При помощи универсальной функции рассмотрим основные данные по таблицам**

In [4]:
# универсальная функция для выполнения предобработки данных
def data_preprocessing(df):
    
    
    # Посмотрим основные данные о DataFrame
    print('Информация о данных (info()):')
    print(df.info())
    print('*' * 100)
    print()
    
    # найдем количество пропусков в процентном соотношении для каждого столбца
    print('Количество пропусков в датафрейме:')
    data_pass = pd.DataFrame(df.isna().mean()*100)
    data_pass = data_pass.rename(columns={0 : '% пропусков'})
    data_pass = data_pass.style.background_gradient('coolwarm').format('{:.1f}%')
    display(data_pass)
    print('*' * 100)
    print()
    
    # Оценим статистические данные
    print('Статистическая информация о данных (describe()):')
    display(df.describe())
    print('*' * 100)
    print()
    
    # Посмотрим на типы данных в DataFrame
    print('Типы данных (dtypes):')
    print(df.dtypes)
    print('*' * 100)
    print()
    
    # Посчитаем количество явных дубликатов, удалим их при наличии
    print('Кол-во дубликатов = ', df.duplicated().sum())
    print('*' * 100)

    print()

### Поработаем с таблицей tos

In [5]:
# взгляем на таблицу
tos.head()

№  1 Поселение                            Населенный пункт 2 Название ТОС  \
0  1  Октябрьский  Городское поселение "Поселок Октябрьский"      «Питомник»   
1  2  Октябрьский  Городское поселение "Поселок Октябрьский"         «Олимп»   
2  3  Октябрьский  Городское поселение "Поселок Октябрьский"      «Заречный»   
3  4  Октябрьский  Городское поселение "Поселок Октябрьский"   «Центральный»   
4  5  Октябрьский  Городское поселение "Поселок Октябрьский"        «Юность»   

  3 Форма регистрации                                  Адрес регистрации  \
0              без ЮЛ    Территория в районе пруда «Питомник» Шаламов Яр   
1              без ЮЛ  В пределах территории спортплощадки по ул. Кор...   
2              без ЮЛ  В пределах территории детской площадки по ул. ...   
3              без ЮЛ  В пределах территории ул. Шевченко – пер. Бели...   
4              без ЮЛ  В пределах территории ул. Первомайская – ул. С...   

   4 Количество зарегистрированных жителей на данной территории  \
0                                               1433              
1                                                563              
2                                                271              
3                                                231              
4                                                148              

   5 Количество членов ТОС  \
0                       25   
1                       19   
2                       16   
3                       35   
4                       15   

   6 Количество активистов ТОС (тех, кто регулярно принимает участие в деятельности ТОС)  \
0                                                  4                                       
1                                                  7                                       
2                                                  2                                       
3                                                  2                                       
4                                                  2                                       

  7 Пол председателя 8 Дата избрания председателем ТОС  \
0                муж               2017-02-08 00:00:00   
1                жен               2021-04-13 00:00:00   
2                жен               2019-04-05 00:00:00   
3                жен               2021-05-11 00:00:00   
4                муж               2021-07-19 00:00:00   

  Место работы, должность председателя ТОС         9 Занятость   \
0                              Безработный          Безработный   
1                              Безработная          Безработный   
2                              Домохозяйка          Домохозяйка   
3           АО "Тандер", директор магазина  Руководитель бизнес   
4                           Мобилизованный       Мобилизованный   

  10 Дата рождения председателя ТОС 11 Активность ТОС  \
0                        1962-07-20          Активный   
1                        1986-09-15          Активный   
2                        1984-12-11          Активный   
3                        1979-09-23        неактивный   
4                        1983-10-08        неактивный   

  12 Партийная принадлежность председателя ТОС  13 Уровень лояльности  \
0                                          б/п                      5   
1                                          б/п                      5   
2                                          б/п                      5   
3                                          б/п                      4   
4                                          б/п                      3   

  14 Проект поданный на конкурс "Решаем вместе" в 2022 году (наименование)  \
0                                                нет                         
1                                                нет                         
2                                                нет                         
3  Поставка и установка детской игровой,\nспортив...                         
4   

In [6]:
# применим новые названия столбцам для удобства работы с ними
tos.columns = [
    '№', 
    'Поселение', 
    'Населенный пункт', 
    'Название ТОС', 
    'Форма регистрации', 
    'Адрес регистрации',
    'Количество зарегистрированных жителей', 
    'Количество членов ТОС', 
    'Количество активистов ТОС',
    'Пол председателя', 
    'Дата избрания', 
    'Место работы председателя', 
    'Занятость',
    'Дата рождения председателя', 
    'Активность ТОС', 
    'Партийная принадлежность', 
    'Уровень лояльности',
    'Проект "Решаем вместе" 2022', 
    'Сумма финансирования проекта "Решаем вместе"', 
    'Проект областного конкурса 2022', 
    'Сумма финансирования областного конкурса',
    'Проект муниципального конкурса 2022', 
    'Сумма финансирования муниципального конкурса',
    'Финансирование председателя', 
    'Основные направления деятельности ТОС', 
    'Наличие соцсети'
]

In [7]:
# исследуем таблицу tos
data_preprocessing(tos)

Информация о данных (info()):
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 199 entries, 0 to 198
Data columns (total 26 columns):
 #   Column                                        Non-Null Count  Dtype         
---  ------                                        --------------  -----         
 0   №                                             199 non-null    int64         
 1   Поселение                                     199 non-null    object        
 2   Населенный пункт                              199 non-null    object        
 3   Название ТОС                                  199 non-null    object        
 4   Форма регистрации                             199 non-null    object        
 5   Адрес регистрации                             199 non-null    object        
 6   Количество зарегистрированных жителей         199 non-null    int64         
 7   Количество членов ТОС                         199 non-null    int64         
 8   Количество активистов ТОС               

****************************************************************************************************

Статистическая информация о данных (describe()):


№  Количество зарегистрированных жителей  Количество членов ТОС  \
count 199.00                                 199.00                 199.00   
mean  101.39                                 329.23                 127.61   
min     1.00                                  20.00                   4.00   
25%    50.50                                 106.00                  25.00   
50%   102.00                                 167.00                  56.00   
75%   151.50                                 314.50                 150.00   
max   201.00                               13587.00                1120.00   
std    58.33                                 979.43                 177.41   

       Количество активистов ТОС     Дата рождения председателя  \
count                     199.00                            199   
mean                        9.81  1975-09-03 08:55:28.643216096   
min                         0.00            1946-01-01 00:00:00   
25%                         3.00            1964-04-07 12:00:00   
50%                         5.00            1979-12-26 00:00:00   
75%                        13.00            1986-01-01 00:00:00   
max                        63.00            2005-06-07 00:00:00   
std                        10.61                            NaN   

       Уровень лояльности  
count              199.00  
mean                 3.85  
min                  1.00  
25%                  3.00  
50%                  4.00  
75%                  5.00  
max                  5.00  
std                  1.11

****************************************************************************************************

Типы данных (dtypes):
№                                                        int64
Поселение                                               object
Населенный пункт                                        object
Название ТОС                                            object
Форма регистрации                                       object
Адрес регистрации                                       object
Количество зарегистрированных жителей                    int64
Количество членов ТОС                                    int64
Количество активистов ТОС                                int64
Пол председателя                                        object
Дата избрания                                           object
Место работы председателя                               object
Занятость                                               object
Дата рождения председателя                      datetime6

In [8]:
# по заданию нам не нужны эти столбцы. Удалю их
tos = tos.drop(['№','Уровень лояльности'], axis=1)

In [9]:
tos.head()

Поселение                            Населенный пункт   Название ТОС  \
0  Октябрьский  Городское поселение "Поселок Октябрьский"      «Питомник»   
1  Октябрьский  Городское поселение "Поселок Октябрьский"         «Олимп»   
2  Октябрьский  Городское поселение "Поселок Октябрьский"      «Заречный»   
3  Октябрьский  Городское поселение "Поселок Октябрьский"   «Центральный»   
4  Октябрьский  Городское поселение "Поселок Октябрьский"        «Юность»   

  Форма регистрации                                  Адрес регистрации  \
0            без ЮЛ    Территория в районе пруда «Питомник» Шаламов Яр   
1            без ЮЛ  В пределах территории спортплощадки по ул. Кор...   
2            без ЮЛ  В пределах территории детской площадки по ул. ...   
3            без ЮЛ  В пределах территории ул. Шевченко – пер. Бели...   
4            без ЮЛ  В пределах территории ул. Первомайская – ул. С...   

   Количество зарегистрированных жителей  Количество членов ТОС  \
0                                   1433                     25   
1                                    563                     19   
2                                    271                     16   
3                                    231                     35   
4                                    148                     15   

   Количество активистов ТОС Пол председателя        Дата избрания  \
0                          4              муж  2017-02-08 00:00:00   
1                          7              жен  2021-04-13 00:00:00   
2                          2              жен  2019-04-05 00:00:00   
3                          2              жен  2021-05-11 00:00:00   
4                          2              муж  2021-07-19 00:00:00   

        Место работы председателя            Занятость  \
0                     Безработный          Безработный   
1                     Безработная          Безработный   
2                     Домохозяйка          Домохозяйка   
3  АО "Тандер", директор магазина  Руководитель бизнес   
4                  Мобилизованный       Мобилизованный   

  Дата рождения председателя Активность ТОС Партийная принадлежность  \
0                 1962-07-20       Активный                      б/п   
1                 1986-09-15       Активный                      б/п   
2                 1984-12-11       Активный                      б/п   
3                 1979-09-23     неактивный                      б/п   
4                 1983-10-08     неактивный                      б/п   

                         Проект "Решаем вместе" 2022  \
0                                                нет   
1                                                нет   
2                                                нет   
3  Поставка и установка детской игровой,\nспортив...   
4                                                нет   

  Сумма финансирования проекта "Решаем вместе"  \
0                                            0   
1                                            0   
2                                            0   
3                     731500 руб. + 38500 руб.   
4                                            0   

  Проект областного конкурса 2022 Сумма финансирования областного конкурса  \
0                             нет                                        0   
1                             нет                                        0   
2                             нет                                        0   
3                             нет                                        0   
4                             нет                                        0   

                 Проект муниципального конкурса 2022  \
0  Приобретение садовой техники и саженцев для ТО...   
1  Приобретение садовой техники для ТОС «Олимп», ...   
2  Приобретение садовой техники и озеленение ТОС ...   
3  Приобретение садовой техники и озеленение ТОС ...   
4                                                нет   

  Сумма финансирования муниципального конкурса 

**Вывод:** В таблице содержится 199 строк и 26 столбцов. Названия столбцов приведены в корректный вид, а лишние столбцы удалены. Пропуски наблюдаются только в столбце "Место работы, должность председателя ТОС" и составляют 1%. Явные дубликаты отсутствуют. Необходимо заменить тип данных у некоторых столбцов и проверить корректность данных. 

### Поработаем с таблицей tos_district

In [10]:
tos_district.head()

№            Поселение  Сокр  Площадь, тыс. кв.км  Население, чел.  \
0  11        Беловское с/п  Бело                60.72             6058   
1  10  Беломестненское с/п  Белм                38.80             6092   
2  13     Бессоновское с/п   Бес               164.23             4777   
3  15  Веселолопанское с/п  ВЛоп                40.89             3593   
4  18      Головинское с/п   Гол                93.21             2362   

   Плотность  
0      99.77  
1     157.01  
2      29.09  
3      87.87  
4      25.34

In [11]:
# исследуем таблицу tos_district
data_preprocessing(tos_district)

Информация о данных (info()):
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 0 to 23
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   №                    24 non-null     int64  
 1   Поселение            24 non-null     object 
 2   Сокр                 24 non-null     object 
 3   Площадь, тыс. кв.км  24 non-null     float64
 4   Население, чел.      24 non-null     int64  
 5   Плотность            24 non-null     float64
dtypes: float64(2), int64(2), object(2)
memory usage: 1.3+ KB
None
****************************************************************************************************

Количество пропусков в датафрейме:


****************************************************************************************************

Статистическая информация о данных (describe()):


№  Площадь, тыс. кв.км  Население, чел.  Плотность
count 24.00                24.00            24.00      24.00
mean  12.50                60.58          7945.33     221.50
std    7.07                35.99          7788.17     289.19
min    1.00                11.61          1034.00      15.52
25%    6.75                40.55          2249.25      26.24
50%   12.50                58.70          5136.00      89.56
75%   18.25                66.44         12437.25     278.46
max   24.00               164.23         30517.00    1161.83

****************************************************************************************************

Типы данных (dtypes):
№                        int64
Поселение               object
Сокр                    object
Площадь, тыс. кв.км    float64
Население, чел.          int64
Плотность              float64
dtype: object
****************************************************************************************************

Кол-во дубликатов =  0
****************************************************************************************************



In [12]:
# по заданию нам не нужны эти столбцы. Удалю их
tos_district = tos_district.drop('№', axis = 1)

**Вывод:** Таблица содержит 24 строки и 6 столбцов. Удалил лишний столбец. Названия столбцов оставил без изменений. Пропуски и дубликаты отсутствуют. Тип данных совпадает 

### Поработаем с таблицей tos_area

In [13]:
tos_area.head()

№                            Район  Площадь, кв.км  Площадь, %  \
0 1.00                      г. Белгород          153.10        0.01   
1 2.00  Старооскольский городской округ         1693.50        0.06   
2 3.00               Белгородский район         1453.86        0.05   
3 4.00                Ракитянский район          900.90        0.03   
4 5.00        Валуйский городской округ         1709.60        0.06   

   Население  Население, %  Городское население  Городское население, %  \
0     392426          0.24            392426.00                    0.37   
1     259811          0.16            223809.00                    0.21   
2     190688          0.12             60805.00                    0.06   
3     117017          0.07             86422.00                    0.08   
4      87944          0.05             47390.00                    0.04   

   Сельское население  Сельское население, %  Плотность  \
0                 NaN                   0.00    2563.20   
1            36002.00                   0.07     153.42   
2           129883.00                   0.24     131.16   
3            30595.00                   0.06     129.89   
4            40554.00                   0.07      51.44   

   Плотность городского населения  Плотность сельского населения  Unnamed: 13  \
0                         2563.20                           0.00          NaN   
1                          132.16                          21.26          NaN   
2                           41.82                          89.34          NaN   
3                           95.93                          33.96          NaN   
4                           27.72                          23.72          NaN   

   Unnamed: 14  Unnamed: 15  Unnamed: 16  
0          NaN          NaN          NaN  
1          NaN          NaN         1.00  
2          NaN          NaN          NaN  
3          NaN          NaN          NaN  
4          NaN          NaN          NaN

In [15]:
tos_area = tos_area.drop(['№','Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15', 'Unnamed: 16'], axis=1)

In [16]:
# исследуем таблицу tos_area
data_preprocessing(tos_area)

Информация о данных (info()):
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23 entries, 0 to 22
Data columns (total 12 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Район                           23 non-null     object 
 1   Площадь, кв.км                  23 non-null     float64
 2   Площадь, %                      22 non-null     float64
 3   Население                       23 non-null     int64  
 4   Население, %                    22 non-null     float64
 5   Городское население             22 non-null     float64
 6   Городское население, %          22 non-null     float64
 7   Сельское население              22 non-null     float64
 8   Сельское население, %           22 non-null     float64
 9   Плотность                       22 non-null     float64
 10  Плотность городского населения  22 non-null     float64
 11  Плотность сельского населения   22 non-null     float64
dtypes: float

****************************************************************************************************

Статистическая информация о данных (describe()):


Площадь, кв.км  Площадь, %  Население  Население, %  \
count           23.00       22.00      23.00         22.00   
mean          2350.27        0.05  140327.57          0.05   
std           5397.64        0.02  333936.12          0.06   
min            153.10        0.01   11563.00          0.01   
25%            886.00        0.03   26061.00          0.02   
50%           1369.00        0.05   36539.00          0.02   
75%           1610.05        0.06   76799.00          0.04   
max          27028.06        0.07 1613767.00          0.24   

       Городское население  Городское население, %  Сельское население  \
count                22.00                   22.00               22.00   
mean              96925.00                    0.05            49781.09   
std              234387.69                    0.08           113924.79   
min                5843.00                    0.00             6409.00   
25%                8248.25                    0.01            14409.25   
50%               16704.00                    0.01            21829.50   
75%               45790.00                    0.04            30262.00   
max             1066175.00                    0.37           547592.00   

       Сельское население, %  Плотность  Плотность городского населения  \
count                  22.00      22.00                           22.00   
mean                    0.05     162.71                          142.34   
std                     0.05     537.89                          541.72   
min                     0.00       6.56                            0.00   
25%                     0.02      18.10                            6.39   
50%                     0.04      26.27                           10.75   
75%                     0.05      75.61                           41.44   
max                     0.24    2563.20                         2563.20   

       Плотность сельского населения  
count                          22.00  
mean                           20.37  
std                            18.07  
min                             0.00  
25%                             9.40  
50%                            16.19  
75%                            23.30  
max                            89.34

****************************************************************************************************

Типы данных (dtypes):
Район                              object
Площадь, кв.км                    float64
Площадь, %                        float64
Население                           int64
Население, %                      float64
Городское население               float64
Городское население, %            float64
Сельское население                float64
Сельское население, %             float64
Плотность                         float64
Плотность городского населения    float64
Плотность сельского населения     float64
dtype: object
****************************************************************************************************

Кол-во дубликатов =  0
****************************************************************************************************



**Вывод:** В таблице содержится 23 строки и 12 столбцов. Ненужные столбцы были удалены, что сделало данные более удобными для анализа. В некоторых ячейках имеются пропуски значений, однако дубликаты отсутствуют. Все названия столбцов корректны и отражают суть представленных данных.